src/vmshepherd_runtime_postgres_driver/__init__.py

```diff
@@ -63,18 +64,23 @@ def reconfigure(self, config):
                  'failed_checks': vms_states['failed_checks']
                }
 
-    async def _acquire_lock(self, name):
+    async def _acquire_lock(self, preset_name):
         await self._assure_connected()
-        async with self._pool.acquire() as con:
-            is_locked = await con.fetchval('select pst_is_locked from preset_states where pst_name=$1 for update', name)
-            if is_locked:
-                return False
-            elif is_locked is None:
-                await con.execute('insert into preset_states (pst_name, pst_is_locked) values ($1, true)', name)
-            else:
-                await con.execute('update preset_states set pst_is_locked=true where pst_name=$1', name)
-        return True
+        try:
+            async with self._pool.acquire() as con:
+                is_locked = await con.fetchval(
+                    'SELECT * FROM lock_preset($1)', preset_name
```

Comment:
In my opinion using of '*' in a SQL in a code, is a antipattern, because when you change the structure of a table, you will need to refactore code also.

Reply:
when you change the structure you will have to change the code whenever you used the `*` or not. Moreover if you want to fetch all fields it's odd to mention them all here.